src/index.tsx

```diff
@@ -105,14 +105,16 @@ startMathJax();
 serviceWorker.register()
   .then((registration) => {
     app.services.serviceWorker = registration;
-
-    if (registration && (registration.waiting || registration.installing)) {
+    const serviceWorkerController = window.navigator.serviceWorker.controller;
+    if (registration && (registration.waiting || registration.installing) && serviceWorkerController) {
       app.store.dispatch(updateAvailable());
     } else if (registration) {
       // For Chrome and Edge registration.waiting and registration.installing
       // is still null for some time after .register()
       registration.addEventListener('updatefound', () => {
-        app.store.dispatch(updateAvailable());
+        if (serviceWorkerController) {
```

Comment:
Do we need this `if` here?

Reply:
Tbh I do not fully understand the purpose of the first `if` statement.
``` const serviceWorkerController = window.navigator.serviceWorker.controller;
    if (registration && (registration.waiting || registration.installing) && serviceWorkerController) {
      app.store.dispatch(updateAvailable());
    }
```
`updatefound` is also getting fierd for new users, so that `if`checks if the `serviceWorker` `controller` is already present (does nothing for new users). 
```
    else if (registration) {
      registration.addEventListener('updatefound', () => {
        if (serviceWorkerController) {
          app.store.dispatch(updateAvailable());
        }
      });
    }
 ```

Reply:
I was also wondering about this. I've left a comment about removing the first if statement here: https://github.com/openstax/unified/issues/1386#issuecomment-741848714 but Tom commented that we need it https://github.com/openstax/unified/issues/1386#issuecomment-745382064
I'm not really sure, but I think there might be the case when the `update` event is fired before we add a listener to the `sw` and this is why we check for `waiting` and `installing`? cc @TomWoodward